test/tests/IniReaderTest.cpp

```diff
@@ -0,0 +1,66 @@
+#include <gtest/gtest.h>
+#include <limits>
+#include "openrct2/config/ConfigEnum.hpp"
+#include "openrct2/config/IniReader.hpp"
+#include "openrct2/core/MemoryStream.h"
+#include "openrct2/core/Util.hpp"
+
+class IniReaderTest : public testing::Test
+{
+protected:
+    static const utf8 predefined[];
+};
+
+static auto Enum_Currency = ConfigEnum<sint32>({});
+
+TEST_F(IniReaderTest, create_empty)
+{
+    MemoryStream * ms = new MemoryStream(0);
+    ASSERT_NE(ms, nullptr);
+    ASSERT_EQ(ms->CanRead(), true);
+    ASSERT_EQ(ms->CanWrite(), true);
+    IIniReader * ir = CreateIniReader(ms);
+    ASSERT_NE(ir, nullptr);
+    ASSERT_EQ(ir->GetBoolean("nobody", true), true);
+    ASSERT_EQ(ir->GetCString("expects", nullptr), nullptr);
+    ASSERT_EQ(ir->GetEnum<sint32>("spanish", 12345, Enum_Currency), 12345);
+    ASSERT_EQ(ir->GetFloat("inquisition", 1.234f), 1.234f);
+    ASSERT_EQ(ir->GetSint32("universal_answer", 42), 42);
+    delete ir;
+    delete ms;
+}
+
+TEST_F(IniReaderTest, read_prepared)
+{
+    MemoryStream * ms = new MemoryStream(predefined, 99);
+    ASSERT_NE(ms, nullptr);
+    ASSERT_EQ(ms->CanRead(), true);
+    ASSERT_EQ(ms->CanWrite(), false);
+    IIniReader * ir = CreateIniReader(ms);
+    ASSERT_NE(ir, nullptr);
+    ASSERT_EQ(ir->ReadSection("doesnt_exist"), false);
+    ASSERT_EQ(ir->ReadSection("bool"), true);
+    // name of section
+    ASSERT_EQ(ir->GetSint32("bool", 42), 42);
+    // value from different section
+    ASSERT_EQ(ir->GetSint32("one", 42), 42);
+    // existing value as different type
+    EXPECT_THROW(ir->GetSint32("boolval", 42), std::invalid_argument);
```

Comment:
I didn't know the convert functions throw this... this will need to be caught in Config.cpp.